### ISSUE TITLE
Required Upstream Infrastructure for SPICE Analysis

### ISSUE THREAD

**Issue by DanRyanIrish** [sunpy/sunraster#176]
To support tools for performing instrument-specific spectral analysis on SPICE data, the following generalized infrastructure must be developed/improved with the following priority:

| Functionality  | Use Case  | Package  | Equivalent/Generalized Infrastructure  | Open Issue/PR  |
| :------------: | :---------: | :--------: | :-------------------------------------: | :--------------: |
|  Fast Non-linear Fitting | Fitting spectral line in all pixels | astropy | Parallelizing non-linear fitting | [astropy#3670](https://github.com/astropy/astropy/issues/3670), [astropy#10985](https://github.com/astropy/astropy/pull/10985) |
|  Update sunraster for ndcube 2.0 | Increase code stability | sunraster |  | [sunraster#177](https://github.com/sunpy/sunraster/issues/177) |
|  Superpixel with method option, e.g. mean, sum etc. | Increase signal-to-noise by degrading spatial, spectral of temporal resolution | ndcube | resampling an NDCube | [ndcube#155](https://github.com/sunpy/ndcube/issues/155), [ndcube#290](https://github.com/sunpy/ndcube/issues/290) |
|  Arithmetic operations | Subtract background | ndcube and/or astropy |  Arithmetic operations | [astropy#11176](https://github.com/astropy/astropy/issues/11176) |

**Comment by ebuchlin** [comment]
Some of these issues have been closed: https://github.com/sunpy/sunraster/issues/177, https://github.com/sunpy/ndcube/issues/155 as done, https://github.com/astropy/astropy/pull/10985 as end of experiment.

**Comment by wtbarnes** [comment]
The last line "arithmetic operations" has also been completed as of sunpy/ndcube#541

**Comment by DanRyanIrish** [comment]
Superpixelling is now possible with ndcube v 2.1.1 via the `NDCube.rebin` method. (sunpy/ndcube#450).  So it seems the only outstanding item is better fitting support in astropy.

**Comment by nabobalis** [comment]
> > the only outstanding item is better fitting support in astropy
> 
> Trivial facepalm

You best fix it and soon.

**Comment by Cadair** [comment]
I have written up this: https://github.com/astropy/astropy/issues/14610 to summarise what's in my head today.

**Comment by Cadair** [comment]
The fitting support is done 🚀

**Comment by ebuchlin** [comment]
Are you referring to the new [Fitting models in parallel with N-dimensional data](https://docs.astropy.org/en/stable/whatsnew/7.0.html#fitting-models-in-parallel-with-n-dimensional-data) in astropy 7.0.0? I now see [PR16696](https://github.com/astropy/astropy/pull/16696) and others about this. Thanks for pushing this into astropy, and you, @astrofrog and others for the hard work!

**Comment by Cadair** [comment]
Yep that's the stuff. You can see an example of using it with SPICE data here: https://github.com/aperiosoftware/parallel-modeling-examples/blob/main/spice/Final_Tutorial.ipynb
